processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -137,6 +139,34 @@ public boolean applyLong(long input)
             };
           }
         }
```

Comment:
I changed the String predicate here to use `Predicates.equalTo()` and replaced the always false predicates with singletons